### PR TITLE
3280 fix tables tests

### PR DESCRIPTION
**Pull request by rafatower**
@juanignaciosl @Kartones can you please review?

What's in this PR:
- Delete many tests that referenced an old endpoint used for tables listing: `v1_tables_url`
- Delete unused `vizzjson` endpoint (I think Alejandro did the very same thing)
- Recover some E2E tests that can still be useful
- Some fixes for stuff I presumably broke while doing the `UserTable` refactor
- Finally add the suite back to the Makefile

### CODE REVIEW

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3142/))

**Comment by Kartones** [comment]
+1

File: spec/requests/api/tables_spec.rb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #encoding: UTF-8
-require_relative '../../acceptance_helper'
+require_relative '../../acceptance_helper' # TODO: is this used?
```
Comment (juanignaciosl) [inline spec/requests/api/tables_spec.rb]: Try deleting it :)

**Comment by juanignaciosl** [comment]
:+1:

File: spec/requests/api/tables_spec.rb (hunk 1 above)
Comment (rafatower) [inline spec/requests/api/tables_spec.rb]: hehehe, it is used but quite crappy. I will adapt these tests to the "new controller style" and then maybe remove that file altogether (capybara deprecated stuff mixed with rack and warden test helpers)

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3155/))

**Pull request merged**